erpnext/crm/report/opportunity_summary_by_sales_stage/opportunity_summary_by_sales_stage.py

```diff
@@ -0,0 +1,274 @@
+# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe import _
+from six import iteritems
+import json
+
+
+def execute(filters=None):
+	return OpportunitySummaryBySalesStage(filters).run()
+
+class OpportunitySummaryBySalesStage(object):
+
+	def __init__(self,filters=None):
+		self.filters = frappe._dict(filters or {})
+
+	def run(self):
+		self.get_columns()
+		self.get_data()
+		self.get_chart_data()
+
+		return self.columns,self.data,None,self.chart
+
+	def get_columns(self):
+		self.columns = []
+
+		if self.filters.get('based_on') == "Opportunity Owner":
+			self.columns.append({
+				'label': _('Opportunity Owner'),
+				'fieldname': 'opportunity_owner',
+				'width':200
+			})
+		if self.filters.get('based_on') == "Source":
+			self.columns.append({
+				'label': _('Source'),
+				'fieldname': 'source',
+				'fieldtype': 'Link',
+				'options':'Lead Source',
+				'width': 200
+			})
+		if self.filters.get('based_on') == "Opportunity Type":
+			self.columns.append({
+				'label': _('Opportunity Type'),
+				'fieldname': 'opportunity_type',
+				'width': 200
+			})
+
+		
+		self.sales_stage_list = frappe.db.get_list("Sales Stage",pluck="name")
+		for sales_stage in self.sales_stage_list:
+			self.columns.append({
+				'label': _(sales_stage),
+				'fieldname': sales_stage,
+				'width':150
+			})
+
+		return self.columns
+
+	def get_data(self):
+		self.data = []
+		if self.filters.get('based_on') == "Opportunity Owner":
+			sql = "_assign"
+			self.get_data_query(sql)
+		if self.filters.get('based_on') == "Source":
+			sql = "source"
+			self.get_data_query(sql)
+		if self.filters.get('based_on') == "Opportunity Type":
+			sql = "opportunity_type"
+			self.get_data_query(sql)
```

Comment:
use this instead of multiple if statements
```
sql = {
  "Opportunity Owner": "_assign",
  "Source": "source",
  "Opportunity Type": "opportunity_type"
}[self.filters.get('based_on')]

self.get_data_query(sql)
```